### PR TITLE
implement curved roads feature

### PR DESCRIPTION
**Pull request by econaxis**
Please read [our contributing guide](https://github.com/graphhopper/graphhopper/blob/master/CONTRIBUTING.md) and note that also your contribution falls under the Apache License 2.0 as outlined there.

Your first contribution should include a change where you add yourself to the CONTRIBUTORS.md file.

### CODE REVIEW

File: core/src/main/java/com/graphhopper/isochrone/algorithm/PillarEdgeResolver.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+import com.graphhopper.storage.Graph;
+import com.graphhopper.storage.NodeAccess;
+import com.graphhopper.util.*;
+import com.graphhopper.util.shapes.GHPoint;
+import com.graphhopper.util.shapes.GHPoint3D;
+
+import java.util.function.Consumer;
+
+public class PillarEdgeResolver implements Consumer<ShortestPathTree.IsoLabel> {
```
Comment (EddyIonescu) [inline core/src/main/java/com/graphhopper/isochrone/algorithm/PillarEdgeResolver.java:16]: Could you add a sentence saying what this does and why it exists?

File: web-bundle/src/main/java/com/graphhopper/resources/SPTResource.java [hunk 2]
```diff
@@ -218,7 +225,24 @@ public Response doGet(
...
+                        // TODO: make sure node/adj_node is correct for coord/prevCoord
+                        label.coordinate = l.node;
+                        label.prevCoordinate = l.prev_node;
+                        isoLabelConsumer.accept(label);
+                    }, graph);
+                    shortestPathTree.search(snap.getClosestNode(), pillarEdgeResolver);
+                } else {
+                    shortestPathTree.search(snap.getClosestNode(), l -> {
+                        IsoLabelWithCoordinates label = isoLabelWithCoordinates(nodeAccess, l);
+                        isoLabelConsumer.accept(label);
```
Comment (EddyIonescu) [inline web-bundle/src/main/java/com/graphhopper/resources/SPTResource.java]: Was wondering will this path ever run - if not could a comment be added to clarify?

**Review by EddyIonescu — COMMENTED** [review]
Looks good so far, just would want to confirm it works in trips-viewer

File: web-bundle/src/main/java/com/graphhopper/resources/SPTResource.java (hunk 2 above)
Comment (econaxis) [reply web-bundle/src/main/java/com/graphhopper/resources/SPTResource.java]: added an optional parameter in the URL params that allow either branch to run

**Review by EddyIonescu — APPROVED** [review]

**Pull request merged**